Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -2097,7 +2097,7 @@ void WebPageProxy::loadRequestWithNavigationShared(Ref<WebProcessProxy>&& proces
             RefPtr protectedThis = weakThis.get();
             if (!protectedThis)
                 return;
-            protectedThis->m_navigationClient->didFailProvisionalNavigationWithError(*protectedThis, legacyEmptyFrameInfo(WTFMove(request)), navigation.ptr(), request.url(), cannotShowURLError(request), nullptr);
+            protectedThis->m_navigationClient->didFailProvisionalNavigationWithError(*protectedThis, legacyEmptyFrameInfo(request.isolatedCopy()), navigation.ptr(), request.url(), cannotShowURLError(request), nullptr);
```

Comment:
Nice catch! `isolatedCopy` does deep copy of the whole object, which we don't need here.
We might just copy the member before the function:
```
auto requestURL = request.url();
auto error = cannotShowURLError(request);
protectedThis->m_navigationClient->didFailProvisionalNavigationWithError(*protectedThis, legacyEmptyFrameInfo(WTFMove(request)), navigation.ptr(), requestURL, error, nullptr);
```